src/sentry/static/sentry/app/components/collapsible.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+
+import Button from 'app/components/button';
+import {t, tn} from 'app/locale';
+
+type CollapseButtonRenderProps = {
+  handleCollapse: () => void;
+};
+
+type ExpandButtonRenderProps = {
+  handleExpand: () => void;
+  numberOfCollapsedItems: number;
+};
+
+type DefaultProps = {
+  maxVisibleItems: number;
+};
+
+type Props = {
+  collapseButton?: (props: CollapseButtonRenderProps) => React.ReactNode;
+  expandButton?: (props: ExpandButtonRenderProps) => React.ReactNode;
+} & DefaultProps;
+
+type State = {
+  collapsed: boolean;
+};
+
+class Collapsible extends React.Component<Props, State> {
+  static defaultProps: DefaultProps = {
+    maxVisibleItems: 5,
+  };
+
+  state: State = {
+    collapsed: true,
+  };
+
+  onCollapseToggle = () => {
+    this.setState(prevState => ({
+      collapsed: !prevState.collapsed,
+    }));
+  };
+
+  renderCollapseButton() {
+    const {collapseButton} = this.props;
+
+    if (typeof collapseButton === 'function') {
+      return collapseButton({handleCollapse: this.onCollapseToggle});
```

Comment:
We should swap the prefixes here, the callback should be `onCollapse` and the method should be `this.handleCollapseToggle`.

Reply:
good catch, will do